File: gps124-gps-navigation/app/src/main/java/com/gpsnavigation/maps/gpsroutefinder/routemap/adapters/OnBoardingAdapter.java
package com.gpsnavigation.maps.gpsroutefinder.routemap.adapters;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.lifecycle.Lifecycle;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import com.gpsnavigation.maps.gpsroutefinder.routemap.fragments.FirstScreenFragment;
import com.gpsnavigation.maps.gpsroutefinder.routemap.fragments.SecondScreenFragment;
import com.gpsnavigation.maps.gpsroutefinder.routemap.fragments.ThirdScreenFragment;

public class OnBoardingAdapter extends FragmentStateAdapter {


    public OnBoardingAdapter(@NonNull FragmentManager fragmentManager, @NonNull Lifecycle lifecycle) {
        super(fragmentManager, lifecycle);
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {

        switch (position) {
            case 1:
                return new SecondScreenFragment();
            case 2:
                return new ThirdScreenFragment();
            default:
                return new FirstScreenFragment();
        }

    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public int getItemCount() {
        // Return the total number of fragments
        return 3; // Or the number of fragments you have
    }
}

File: gps160-beauty-camera-photo-edi/app/src/main/java/com/lockerroom/face/activities/onboardingScreen/OnBoardingAdapter.kt
package com.lockerroom.face.activities.onboardingScreen

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.lifecycle.Lifecycle
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.lockerroom.face.activities.onboardingScreen.onboarding_fragments.OnBoardingFragmentOne
import com.lockerroom.face.activities.onboardingScreen.onboarding_fragments.OnBoardingFragmentThree
import com.lockerroom.face.activities.onboardingScreen.onboarding_fragments.OnBoardingFragmentTwo

class OnBoardingAdapter(fragmentManager: FragmentManager, lifecycle: Lifecycle) :
    FragmentStateAdapter(fragmentManager, lifecycle) {


    override fun getItemCount(): Int {
        return 3
    }

    override fun createFragment(position: Int): Fragment {
        return if (position == 0) {
            OnBoardingFragmentOne()
        } else if (position == 1) {
            OnBoardingFragmentTwo()
        } else {
            OnBoardingFragmentThree()
        }
    }
}